src/bsp/src/posix/src/SpiStmMock.cpp

```diff
@@ -0,0 +1,89 @@
+#include "SpiStmMock.h"
+#include "Task.h"
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+SpiStmMock::SpiStmMock(ILogger& logger, const char* address, int port) : m_logger(logger) {
+    m_socket = -1;
+    m_address.sin_family = AF_INET;
+    m_address.sin_port = htons(port);
+    m_address.sin_addr.s_addr = (inet_addr(address));
+
+    while (m_socket < 0) {
+        m_logger.log(LogLevel::Error, "Attempting connection to HiveMind");
+        this->connect();
+        Task::delay(500);
+    }
+}
+
+SpiStmMock::~SpiStmMock() noexcept { close(); }
```

Comment:
Is there a reason for the noexcept? I think exceptions are disabled on the build anyway  no?

Reply:
Probably an auto-complete issue, I'll remove it